receipt_dynamo/receipt_dynamo/entities/receipt_letter.py

```
@@ -63,53 +50,46 @@ class ReceiptLetter(
 
     def __post_init__(self) -> None:
         """Validate and normalize initialization arguments."""
+        # Validate receipt_id
         if not isinstance(self.receipt_id, int):
             raise ValueError("receipt_id must be an integer")
         if self.receipt_id <= 0:
             raise ValueError("receipt_id must be positive")
 
+        # Validate line_id
         if not isinstance(self.line_id, int):
             raise ValueError("line_id must be an integer")
```

Ruff (0.14.13)

Prefer `TypeError` exception for invalid type

(TRY004)